source/common/modules/markdown-editor/commands/autocorrect.ts

```diff
@@ -35,12 +35,20 @@ const startChars = ' ([{-–—\n\r\t\v\f'
  */
 function posInProtectedNode (state: EditorState, pos: number): boolean {
   const node = syntaxTree(state).resolve(pos, 0)
+
+  // Having the cursor at the end of the HorizontalRule '---' is not considered inside the HorizontalRule node
+  // So we check the previous position to see if it is in a HorizontalRule node
+  const checkYAMLFrontmatterNode = syntaxTree(state).resolve(pos - 1, 0)
```

Comment:
The second resolve parameter indicates a "side" which, if I read the docs correctly, indicate into which direction it should look to find a node. It seems safer to me to use that parameter – if it does what you desire. Could you test this out? (Refer to the CM docs for a description of that parameter)